### PR TITLE
[BUG] Memory leak issue ,Clean Up handling,Preserve handling during compaction

### PR DESCRIPTION
**Pull request by ravikiran23**
1. clean up handling is done in compaction flow.
2. Fixed the memory leak issue in query execution flow.
3. Corrected the preserve segment to exclude the already merged segment from the preserve.

### CODE REVIEW

File: integration/spark/src/main/java/org/carbondata/spark/merger/CarbonDataMergerUtil.java [hunk 1]
```diff
@@ -588,32 +588,25 @@ private static boolean isMergedSegment(String segName) {
...
+      int removingIndex = validList.size() - 1;
+      for (int i = 0; i < numberOfSegToBeRetained; i++) {
 
-          if ((segNumber1 - segNumber2) < 0) {
-            return -1;
-          } else if ((segNumber1 - segNumber2) > 0) {
-            return 1;
-          }
-          return 0;
+        // remove last segment if it is not a merged segment
```
Comment (gvramana) [inline integration/spark/src/main/java/org/carbondata/spark/merger/CarbonDataMergerUtil.java]: Should break logic when merge segment found or preserve merge segments also.

**Comment by ravikiran23** [comment]
retest this please

**Comment by ravikiran23** [comment]
retest this please

**Comment by ravikiran23** [comment]
retest this please

**Comment by ravikiran23** [comment]
retest this please

**Comment by ravikiran23** [comment]
retest this please

File: integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 2]
```diff
@@ -579,7 +605,7 @@ object CarbonDataRDDFactory extends Logging {
       // Deleting the any partially loaded data if present.
       // in some case the segment folder which is present in store will not have entry in status.
       // so deleting those folders.
-      CarbonLoaderUtil.deletePartialLoadDataIfExist(carbonLoadModel)
+      CarbonLoaderUtil.deletePartialLoadDataIfExist(carbonLoadModel, false)
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: add try catch here also

**Pull request merged**